app/src/main/java/org/glucosio/android/tools/GlucoseRanges.java

```diff
@@ -33,23 +35,48 @@
     private int customMin;
     private int customMax;
 
+    @VisibleForTesting
+    protected boolean isInUnitTests() {
+        return false;
+    }
+
+    @VisibleForTesting
+    public void setPreferredRange(String preferredRange) {
+        this.preferredRange = preferredRange;
+    }
+
+    @VisibleForTesting
+    public void setCustomMin(int customMin) {
+        this.customMin = customMin;
+    }
+
+    @VisibleForTesting
+    public void setCustomMax(int customMax) {
+        this.customMax = customMax;
+    }
+
+
+    public GlucoseRanges() {
+        //default empty constructor
+    }
+
     public GlucoseRanges(Context context) {
         this.mContext = context;
-        dB = new DatabaseHandler(mContext);
-        this.preferredRange = dB.getUser(1).getPreferred_range();
-        if (preferredRange.equals("Custom range")) {
-            this.customMin = dB.getUser(1).getCustom_range_min();
-            this.customMax = dB.getUser(1).getCustom_range_max();
+        if (!isInUnitTests()) {
+            dB = new DatabaseHandler(mContext);
```

Comment:
Well, I discovered some issues when I tried to use Robolectric with Realm database. These are known, but not fixed, and you would find them on robolectric or realm repository site. 

I tried also to use mockito to avoid robolectric-realm conflicts, but with no result.

As I hadn't found any other solution, I made this one with `isUnitTests()` method to get rid of using any database. This solution isn't wild as you may think, because it is used to avoid conflict using Robolectric with other libraries (so not only realm) like LeakCanary.

If you could write another working solution to this problem, I would glad to check it.

Reply:
Check please `A1CCalculatorPresenterTest`. We mock `DatabaseHandler` and inject it to presenter.

Reply:
@emartynov ok, I will check it and I hope I would fix it